hivemind/p2p/p2p_daemon.py

```diff
@@ -375,15 +379,22 @@ async def add_protobuf_handler(
         handler: Callable[
             [Union[TInputProtobuf, TInputStream], P2PContext], Union[Awaitable[TOutputProtobuf], TOutputStream]
         ],
-        input_protobuf_type: type,
+        input_protobuf_type: Message,
         *,
         stream_input: bool = False,
+        stream_output: bool = False,
     ) -> None:
         """
         :param stream_input: If True, assume ``handler`` to take ``TInputStream``
                              (not just ``TInputProtobuf``) as input.
+        :param stream_output: If True, assume ``handler`` to return ``TOutputStream``
+                              (not ``Awaitable[TOutputProtobuf]``).
         """
 
+        if not stream_input and not stream_output:
+            await self._add_protobuf_unary_handler(name, handler, input_protobuf_type)
```

Comment:
I'm afraid this might lead to copy-pasting some code. Should we should vote on this.

Reply:
@mryab Since this PR provides an optimization for unary-unary handlers, I think it's reasonable to enable it when possible instead of having two separate APIs for unary handlers (non-optimized handlers added with `add_protobuf_handler()` and optimized handlers added with `add_protobuf_unary_handler()`).

Also, please note that currently this function are not called directly by the user (though it's possible in future) but called inside the `Servicer` (that just sets `stream_input`/`stream_output` flags depending on whether the request/response types are `AsyncIterator` or not).

Reply:
Sounds reasonable, let's keep it as is then. Thanks for an explanation!